src/dmd/parse.d

```diff
@@ -2154,6 +2163,13 @@ final class Parser(AST) : Lexer
                             cppmangle = token.value == TOK.class_ ? CPPMANGLE.asClass : CPPMANGLE.asStruct;
                             nextToken();
                         }
+                        else if (token.value == TOK.leftParentheses) // extern(C++, (StringExp))
+                        {
+                            nextToken();
+                            cppMangleOnly = true;
+                            identExp = parseCondExp();
```

Comment:
My parser knowledge is a bit limited, but shouldn't this be an assignExp?

Reply:
How would an assign expression be valid here? I think `?:` is the highest level grammar applicable in this context...